internal/helm/repository/repository.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"bytes"
+
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+// Downloader is used to download a chart from a remote Helm repository or OCI registry.
+type Downloader interface {
+	// GetChartVersion returns the repo.ChartVersion for the given name and version
+	// from the remote repository.ChartRepository.
+	GetChartVersion(name, version string) (*repo.ChartVersion, error)
+	// DownloadChart downloads a chart from the remote Helm repository or OCI registry.
+	DownloadChart(chart *repo.ChartVersion) (*bytes.Buffer, error)
+	// Clean removes all temporary files created by the downloader, caching the files if the cache is configured,
+	// and calling garbage collector to remove unused files.
+	Clean() []error
```

Comment:
I got into a confusion with `Clean()` in this `Downloader` interface and `Clear()` in the `DependencyManager`. [Clean & Clear](https://en.wikipedia.org/wiki/Clean_%26_Clear) :smile: 
It may be just simpler for searching and remembering to call everything the same. Maybe just `Clear()`.

Reply:
While at it, can we also consider replacing error slice with just error and the implementations can use the `kerrors.Aggregate()` to put the errors together?
There may be good reasons to not do it. Just wondering.